src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
I didn't add that functionality because PHP doesn't run as a service, after it has completed execution the memory is cleared and starts fresh. There are probably only a few edge cases where that would be of value. I could add a method to allow the developer to invoke the inactiveNodes being merged back into the pool. Not sure if it would add much value though.